common/sai.py

```diff
@@ -214,776 +265,80 @@ def get_obj_attr_type(sai_obj_type, sai_obj_attr):
-                    E.g.:
-                    [
-                        {
-                            "bvid"      : vlan_oid,
-                            "mac"       : "00:00:00:00:00:01",
-                            "switch_id" : self.sw_oid
-                        },
-                        {...}
-                    ]
-            attrs (list): The list of objects' attributes, one attribute per object.
-                    In case just one attribute provided, all objects
-                    will be updated with the provided value of this attribute.
-                    E.g.:
-                    [
-                        "SAI_FDB_ENTRY_ATTR_TYPE",           "SAI_FDB_ENTRY_TYPE_STATIC",
-                        "SAI_FDB_ENTRY_ATTR_TYPE",           "SAI_FDB_ENTRY_TYPE_DYNAMIC",
-                        ...
-                    ]
-            do_assert (bool): Assert that the bulk set operation succeeded.
-
-        Usage example:
-            bulk_set(SaiObjType.FDB_ENTRY, [key1, key2, ...], [attr1, attr2, ...])
-            bulk_set(SaiObjType.FDB_ENTRY, [key1, key2, ...], [attr])
-
-        Returns:
-            The tuple with two elements.
-            The first element contains bulk set operation status:
-                * "SAI_STATUS_SUCCESS" on success when all objects were updated;
-                * "SAI_STATUS_FAILURE" when any of the objects fails to update;
-            The second element contains the list of statuses of each individual object
-            set attribute result.
-        '''
-        assert (type(obj) == SaiObjType) or (type(obj) == str and obj.startswith("SAI_OBJECT_TYPE_"))
-        assert len(keys) == len(attrs) or len(attrs) == 1
-
-        key = "SAI_OBJECT_TYPE_" + obj.name if type(obj) == SaiObjType else obj
-        key = key + ":" + str(len(keys))
-
-        str_attr = ""
-        if (len(attrs) == 1):
-            str_attr = self.__bulk_attr_serialize(attrs[0])
-
-        values = []
-        for i, _ in enumerate(keys):
-            k = keys[i]
-            if type(k) != str:
-                k = json.dumps(k).replace(" ", "")
-            values.append(k)
-            if (len(attrs) > 1):
-                str_attr = self.__bulk_attr_serialize(attrs[i])
-            values.append(str_attr)
-
-        status = self.operate(key, json.dumps(values), "Sbulkset")
-
-        status[1] = status[1].decode("utf-8")
-        status[1] = json.loads(status[1])
-        entry_status = []
-        for i, v in enumerate(status[1]):
-            if i % 2 == 0:
-                entry_status.append(v)
-
-        status[2] = status[2].decode("utf-8")
-
-        if do_assert:
-            print(entry_status)
-            assert status[2] == 'SAI_STATUS_SUCCESS'
-            return status[2], entry_status
-
-        return status[2], entry_status
-
-    def get_by_type(self, obj, attr, attr_type, do_assert = True):
+    def get_by_type(self, obj, attr, attr_type):
```

Comment:
Do we use it for Thrift client? I think its Redis specific and should be in sai_redis_client..